### PR TITLE
Update Rust dependencies (Core, Bindings, Build CLI) 

### PR DESCRIPTION
**Pull request by AmmarAbouZor**
This PR updates the rust dependencies among Chipmunk libraries and tools

Only [pcap-parser](https://crates.io/crates/pcap-parser) and [etherparse](https://crates.io/crates/etherparse) couldn't be updated because they introduced breaking changes on their API.

Update:
* Clippy fixes for Rust version 1.83.0 are included in this PR
* PR rebased on master after merging pcap dependencies updates

### CODE REVIEW

**Comment by AmmarAbouZor** [comment]
Hi @DmitryAstafyev 
I've rebased this PR on the top of master and added clippy fixes for Rust version 1.83

File: application/apps/indexer/sources/Cargo.toml [hunk 1]
```diff
@@ -23,7 +23,7 @@ serde = { workspace = true, features = ["derive"] }
 uuid = { workspace = true , features = ["serde", "v4"] }
 regex.workspace = true
 lazy_static.workspace = true
-shellexpand = "3.0.0"
+shellexpand = "3.1.0"
```
Comment (DmitryAstafyev) [inline application/apps/indexer/sources/Cargo.toml:26]: @AmmarAbouZor can we switch it just to "3.1"?

File: application/apps/indexer/session/Cargo.toml [hunk 2]
```diff
@@ -16,20 +16,20 @@ indexer_base = { path = "../indexer_base" }
 lazy_static.workspace = true
 log.workspace = true
 merging = { path = "../merging" }
-mime_guess = "2.0.4"
+mime_guess = "2.0"
 parsers = { path = "../parsers" }
 processor = { path = "../processor" }
-rustc-hash = "1.1"
+rustc-hash = "2"
```
Comment (DmitryAstafyev) [inline application/apps/indexer/session/Cargo.toml:22]: Would it be better to use the same approach as for others "x.x"

File: application/apps/indexer/Cargo.toml [hunk 3]
```diff
@@ -34,8 +34,8 @@ rand = "0.8"
 dirs = "5.0"
 uuid = "1.3"
 grep-searcher = "0.1"
-tempfile = "3.10.0"
-env_logger = "0.10"
+tempfile = "3"
```
Comment (DmitryAstafyev) [inline application/apps/indexer/Cargo.toml:37]: Would it be better to use the same approach as for others "x.x"

File: application/apps/indexer/Cargo.toml [hunk 4]
```diff
@@ -19,8 +19,8 @@ members = [
 log = "0.4"
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
-thiserror = "1.0"
-lazy_static = "1.4"
+thiserror = "2"
```
Comment (DmitryAstafyev) [inline application/apps/indexer/Cargo.toml:22]: Would it be better to use the same approach as for others "x.x"

**Review by DmitryAstafyev — APPROVED** [review]
With a couple of tiny notes

**Pull request merged**